packages/file_selector/file_selector_web/lib/file_selector_web.dart

```diff
@@ -0,0 +1,205 @@
+  }) async =>
+      null;
+
+  /// Load Helper
+  Future<List<XFile>> _loadFileHelper(
+      bool multiple, List<XTypeGroup> acceptedTypes) {
+    final acceptedTypeString = _getStringFromFilterGroup(acceptedTypes);
+
+    final FileUploadInputElement element =
+        createFileInputElement(acceptedTypeString, multiple);
+
+    _target.children.clear();
+    addElementToContainerAndClick(_target, element);
+
+    return getFilesWhenReady(element);
+  }
+
+  /// Default constructor, initializes _target to a DOM element that we can use
+  /// to host HTML elements.
+  /// overrides parameter allows for testing to override functions
+  FileSelectorPlugin({
+    @visibleForTesting FileSelectorPluginTestOverrides overrides,
+  }) : _overrides = overrides {
+    _target = ensureInitialized(_kFileSelectorInputsDomId);
+  }
+
+  /// Registers this class as the default instance of [FileSelectorPlatform].
+  static void registerWith(Registrar registrar) {
+    FileSelectorPlatform.instance = FileSelectorPlugin();
+  }
+
+  void _verifyXTypeGroup(XTypeGroup group) {
+    if (group.extensions == null &&
+        group.mimeTypes == null &&
+        group.webWildCards == null) {
+      StateError(
+          "This XTypeGroup does not have types supported by the web implementation of loadFile.");
+    }
+  }
+
+  /// Convert list of XTypeGroups to a comma-separated string
+  String _getStringFromFilterGroup(List<XTypeGroup> acceptedTypes) {
+    List<String> allTypes = List();
+    for (XTypeGroup group in acceptedTypes ?? []) {
+      _verifyXTypeGroup(group);
+
+      for (String mimeType in group.mimeTypes ?? []) {
+        allTypes.add(mimeType);
+      }
+      for (String extension in group.extensions ?? []) {
+        String ext = extension;
+        if (ext.isNotEmpty && ext[0] != '.') {
+          ext = '.' + ext;
+        }
+
+        allTypes.add(ext);
+      }
+      for (String webWildCard in group.webWildCards ?? []) {
+        allTypes.add(webWildCard);
+      }
+    }
+    return allTypes?.where((e) => e.isNotEmpty)?.join(',') ?? '';
+  }
+
+  /// Creates a file input element with only the accept attribute
+  @visibleForTesting
+  FileUploadInputElement createFileInputElement(
+      String accepted, bool multiple) {
+    if (_hasTestOverrides && _overrides.createFileInputElement != null) {
+      return _overrides.createFileInputElement(accepted, multiple);
+    }
+
+    final FileUploadInputElement element = FileUploadInputElement();
+    if (accepted.isNotEmpty) {
+      element.accept = accepted;
+    }
+    element.multiple = multiple;
+
+    return element;
+  }
+
+  List<XFile> _getXFilesFromFiles(List<File> files) {
+    List<XFile> xFiles = List<XFile>();
+
+    final timeZoneOffset = DateTime.now().timeZoneOffset;
+
+    for (File file in files) {
+      String url = Url.createObjectUrl(file);
+      String name = file.name;
+      int length = file.size;
+      DateTime modified = file.lastModifiedDate.add(timeZoneOffset);
```

Comment:
is `file.lastModifiedDate` in UTC always?

Reply:
Yes, it was converting my local files to UTC. The timeZoneOffset fixed that

Reply:
I just realized that `lastModifiedDate` is deprecated, and we should use `lastModified`:

https://developer.mozilla.org/en-US/docs/Web/API/File/lastModified

vs

https://developer.mozilla.org/en-US/docs/Web/API/File/lastModifiedDate

Reply:
Good news, it appears that `lastModified` doesn't convert my local files to UTC, so this is easier anyway